tests/suite/graceful_recovery_test.go

```diff
@@ -0,0 +1,355 @@
+package suite
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/batch/v1"
+	coordination "k8s.io/api/coordination/v1"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+const (
+	nginxContainerName = "nginx"
+	ngfContainerName   = "nginx-gateway"
+)
+
+// Since checkContainerLogsForErrors may experience interference from previous tests (as explained in the function
+// documentation), this test is recommended to be run separate from other nfr tests.
```

Comment:
When you say run separate, does that mean we can't run this in the pipeline with the other NFR tests? Can we just uninstall and reinstall NGF like we do in others (also see the scale test PR which does it)?

Reply:
Hmm, yea that would be a better approach. I'll add in a teardown and setup in the BeforeAll.